optimum/habana/transformers/generation/utils.py

```diff
@@ -1894,6 +1894,10 @@ def _greedy_search(
                 self._pad_past_key_values(model_kwargs)
                 model_kwargs["pad_done"] = True
 
+        # Clear cache is both are True
+        if model_kwargs.get("use_hpu_graphs", False) and model_kwargs.get("limit_hpu_graphs", False):
```

Comment:
Can we rewrite the condition to safe-guard other test as it is a big change and probably degrade the "First token latency tests"